nsls2forge_utils/all_feedstocks.py

```diff
@@ -0,0 +1,109 @@
+'''
+This code is a rework of
+https://github.com/regro/cf-scripts/blob/master/conda_forge_tick/all_feedstocks.py
+for use by nsls-ii-forge's own auto-tick bot.
+'''
+import github3
+import os
+import datetime
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def get_all_feedstocks_from_github(organization, username='', password=''):
+    '''
+    Gets all public feedstock repository names from the GitHub organization
+    (e.g. nsls-ii-forge).
+
+    Parameters
+    ----------
+    organization: str
+        Name of organization on GitHub.
+    username: str, optional
+        Name of user on GitHub for authentication.
+        Uses environment variables if not specified.
+    password: str, optional
+        Password of user on GitHub for authentication.
+        Uses environment variables if not specified.
+
+    Returns
+    -------
+    names: list
+        List of repository names that end with '-feedstock' (stripped).
+    '''
+    if username == '':
+        gh = github3.login(os.environ['USERNAME'], os.environ['PASSWORD'])
+    else:
+        gh = github3.login(username, password)
```

Comment:
The call `github.login()` can be made once after figuring out values for the variables is done within the `if...else` condition above. Just a suggestion to make a call only once.